File: backend/src/main/java/com/back/domain/chat/dto/response/ChatRoomResponseDto.java
package com.back.domain.chat.dto.response;

import com.back.domain.chat.entity.ChatRoom;
import com.back.domain.member.entity.Member;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record ChatRoomResponseDto(
        Long id,
        LocalDateTime createdAt,
        Long firstMemberId,
        Long secondMemberId
) {
    public static ChatRoomResponseDto from(ChatRoom chatRoom) {
        return ChatRoomResponseDto.builder()
                .id(chatRoom.getId())
                .createdAt(chatRoom.getCreatedAt())
                .firstMemberId(chatRoom.getFirstMember().getId())
                .secondMemberId(chatRoom.getSecondMember().getId())
                .build();
    }
}

File: backend/src/main/java/com/back/domain/chat/dto/response/ChatRoomResponseDto.kt
package com.back.domain.chat.dto.response

import com.back.domain.chat.entity.ChatRoom
import java.time.LocalDateTime

data class ChatRoomResponseDto(
    val id: Long?,
    val createdAt: LocalDateTime?,
    val firstMemberId: Long?,
    val secondMemberId: Long?
) {
    companion object {
        fun from(chatRoom: ChatRoom): ChatRoomResponseDto {
            return ChatRoomResponseDto(
                id = chatRoom.id,
                createdAt = chatRoom.createdAt,
                firstMemberId = chatRoom.firstMember.id,
                secondMemberId = chatRoom.secondMember.id
            )
        }
    }
}